lib/xlat_tables_v2/aarch32/xlat_tables_arch.h

```diff
@@ -51,13 +51,13 @@
 
 # error "PLAT_VIRT_ADDR_SPACE_SIZE is too big."
 
-#elif PLAT_VIRT_ADDR_SPACE_SIZE > (1 << L1_XLAT_ADDRESS_SHIFT)
+#elif PLAT_VIRT_ADDR_SPACE_SIZE > (1ULL << L1_XLAT_ADDRESS_SHIFT)
```

Comment:
That depends on whether the firmware is going to support or not 16KB and 64KB pages at some point, as it means that the shift for a L1 entry would be 36 or 42 respectively.